ts/src/index.ts

```diff
@@ -112,27 +108,56 @@ function initConfigLocal(config: Config): ProfilerConfig {
   return mergedConfig;
 }
 
+async function getGcpMetadata(
+  retriesLeft: number,
+  backoffMillis: number
+): Promise<[string | undefined, string | undefined, string | undefined]> {
+  try {
+    return await Promise.all([
+      gcpMetadata.project('project-id'),
+      gcpMetadata.instance('name'),
+      gcpMetadata.instance('zone'),
+    ]);
+  } catch (err) {
+    if (retriesLeft > 0) {
+      await delay(backoffMillis);
+      return getGcpMetadata(retriesLeft - 1, backoffMillis);
```

Comment:
Can we make it a loop rather than recursion?

Also, it seems that all three calls to the metadata server will be repeated if one of them fails? Doesn't seem right if so.

Reply:
Done.